sktime/forecasting/base/adapters/_statsforecast.py

```diff
@@ -57,19 +54,92 @@ def _fit(self, y, X=None, fh=None):
             The forecasting horizon with the steps ahead to to predict.
             Required (non-optional) here if self.get_tag("requires-fh-in-fit")==True
             Otherwise, if not passed in _fit, guaranteed to be passed in _predict
-        X : optional (default=None)
-            guaranteed to be of a type in self.get_tag("X_inner_mtype")
+        X :  sktime time series object, optional (default=None)
+            guaranteed to be of an mtype in self.get_tag("X_inner_mtype")
             Exogeneous time series to fit to.
 
         Returns
         -------
         self : reference to self
         """
+        del fh  # avoid being detected as unused by ``vulture`` like tools
+
         self._forecaster = self._instantiate_model()
-        self._forecaster.fit(y.values, X.values if X is not None else X)
+
+        y_fit_input = y.to_numpy(copy=False)
+        X_fit_input = X.to_numpy(copy=False) if X is not None else X
+
+        self._forecaster.fit(y_fit_input, X=X_fit_input)
 
         return self
 
+    def _predict_in_or_out_of_sample(self, fh, fh_type, X=None, levels=None):
```

Comment:
I wonder if point and interval predictions could be separated better here, though I can't see a good way given how intertwined it is in `statsforecast`.

Reply:
Do you mean ask for only point or only interval predictions? I don't think that'd be possible, given asking for interval will always give point in `statsforecast`.

OR

What was there in old adapter was for each level, a new call was made to `predict`. Are you talking about this? I felt that was unnecessary comuputation because it supports prediction for multiple levels simultaneously.